### ISSUE TITLE
Symbol linking disambiguation help in warnings

### ISSUE THREAD

**Issue by ktoso** [apple/swift-docc#346]
### Feature Name

Symbol linking disambiguation help in warnings

### Description

The adding of pretty arbitrary `-9ytjf` disambiguation suffixes to symbols is nearly impossible to guess right without using Xcode. And a large part of the server and other ecosystems are not using Xcode to develop their libraries, therefore the CLI experience must be improved so that docc is usable, regardless of IDE choice.

E.g.

```
``activateSingleton()-9ytjf``
```

rather than

```
``activateSingleton()``
```

for targeting

```
public protocol ClusterSingleton: DistributedActor where ActorSystem == ClusterSystem {
    func activateSingleton() async
}
```

is nearly impossible to discover for non-Xcode developers.

---

Instead, we should offer additional notes accompanying the missing symbol warning; 

```
DistributedActors.md:74:3: warning: Topic reference 'VersionVector' couldn't be resolved. No local documentation matches this reference.

// SOMETHING LIKE: 
note: Did you mean one of these:
 - activateSingleton()-9ytjf - ≤full symbol name here≥
```

### Motivation

docc documentation must be editable regardless of source editor of choice of developers. We cannot rely only on autocompletion to form correct symbol links.

### Importance

This is important for ecosystems where using Xcode is not assumed. A lot of the Swift on Server ecosystem developers who write a lot of great documentation and want to be using docc, are prevented from adopting it entirely since they must use Xcode to form correct links. Some of them may be using Linux, so just having an Xcode "handy" to write docc links is also not an option.

**Comment by franklinsch** [comment]
This would be really nice indeed. @d-ronnqvist I believe the new link resolution infrastructure you're building would allow for these kinds of diagnostics?

**Comment by d-ronnqvist** [comment]
Yes, https://github.com/apple/swift-docc/pull/337 adds a warning that lists all of the options when a link is ambiguous. See for example [this test assertion](https://github.com/apple/swift-docc/pull/337/files#diff-59466e1c52cca605e6180d885f5559e6a3262ec1fd6a9cdbdf795ce97b8deec1R561)

The link resolution code doesn't yet have a way to respond with full diagnostics (only the error message strings) so these suggestions wouldn't yet be structured as notes or fixits but the information will be in the warning text.

**Comment by ktoso** [comment]
That looks pretty okey -- would you be able to make this multi-line so it's more "human eye on terminal"-friendly?

**Comment by d-ronnqvist** [comment]
Yes, that would be a fairly small change to the error message. I also think that I need to get more information about the symbol's declarations. There are cases when one of the collisions is defined in a protocol where the colliding declarations would look identical. 

I'm a bit confused by how issues and pull requests are linked together but I do not consider https://github.com/apple/swift-docc/pull/337 to be enough to close this issue. I only meant to related them as an indication that it will provide meaningful improvement for this issue.

**Comment by ktoso** [comment]
Cool, thanks! Yeah more information may be necessaary.

> I'm a bit confused by how issues and pull requests are linked together but I do not consider https://github.com/apple/swift-docc/pull/337 to be enough to close this issue. I only meant to related them as an indication that it will provide meaningful improvement for this issue.

The general way to link PR to issue is to mention issue number in PR review; that's enough.

If you want to automatically resolve an issue when a PR is merged do "resolves #...." in any comment and that'll do it.

**Comment by d-ronnqvist** [comment]
I’m reopening this since there’s still more work to be done. I didn’t mean for it to be closed when I merged the first PR.

**Comment by d-ronnqvist** [comment]
@ktoso would the diagnostic formatting improvements in https://github.com/apple/swift-docc/pull/535 address the remaining issue about presenting the disambiguation help on the command line?

**Comment by d-ronnqvist** [comment]
On main, a link disambiguation warning is displayed like this (I introduced this specific warning as an example):

```
warning: 'translateToFileSpace(_:)' is ambiguous at '/SymbolKit/SymbolGraph/LineList'
   --> ../SymbolGraph/LineList/LineList.swift:173:52-173:76
171 |          Translate a range from its *local space* to this line list's *file space*.
172 | 
173 +          This method calls the overload ``LineList/translateToFileSpace(_:)`` on each bound of the `range`.
    |                                                                            ├─suggestion: Insert '9dlzx' for'func translateToFileSpace(_ position: SourceRange.Position) -> SourceRange.Position?'
    |                                                                            ╰─suggestion: Insert '4tk1b' for'func translateToFileSpace(_ range: SourceRange) -> SourceRange?'
174 | 
175 |          - Note: A range in its local space may cross over documentation comment markers in file space. For example:
```

Feel free to reopen this issue and provide additional information if that doesn't resolve this issue.
